src/sc/proofverifier.cpp

```diff
@@ -46,129 +44,209 @@ void CScProofVerifier::LoadDataForCertVerification(const CCoinsViewCache& view,
         bt.amount = btout.nValue;
     }
 
-    certData.quality = scCert.quality; //Currently quality not yet accounted for in proof verifier
-    if (sidechain.fixedParams.constant.is_initialized())
-        certData.constant = sidechain.fixedParams.constant.get();
-    else
-        certData.constant = CFieldElement{};
+    for (auto entry: scCert.vFieldElementCertificateField)
+    {
+        CFieldElement fe{entry.getVRawData()};
```

Comment:
Per my understanding `entry.getVRawData()` returns raw bytes of custom fields, that are not exactly a field element bytes, but can be less. So `CFieldElement fe{entry.getVRawData()}` leads to the invalid FieldElement and moreover asserts inside `CFieldElement` constructor.
I think we need to use `entry.GetFieldElement` method.

Reply:
It is true that the CFieldElement obj stores internally the raw representation of data, but when it is used for input to the prover, its deserialized version is used via CFieldElement()::GetFieldElement() method.

Reply:
I think, yes.
The important thing is that CFieldElement stores exactly 32 bytes of raw data.
From other side `FieldElementCertificateField` stores up to 32 bytes of raw data.

For example, it can store only 4 bytes representing some int value.
So, `FieldElementCertificateField::getVRawData()` will return vector with the length 4.
The expected behavior is to return the full field element 32 bytes and then use them to create a `CFieldElement` obj.

Reply:
Fixed by implementing a new method called getExtendedRawData() that always returns 32 bytes (eventually padding). The other approach of using GetFieldElement was not implementable since there are cases in which the fixed params are not immediately available (e.g. when updating the cumulative commitment tree).

In the future, we may consider storing the custom fields as 32 bytes field elements even though the real data is smaller; would it be ok or do we need to save as much memory as we can on mainchain side? @albertog78 

In any case this activity should be performed together with the refactoring of the part related to the conversion of data for usage with the CCTP Lib interfaces (e.g. to avoid using `wrappedFieldPtr` in the `CScProofVerifier` and `CScAsyncProofVerifier`).